test/Microsoft.AspNet.Mvc.Razor.Host.Test/MvcRazorHostTest.cs

```diff
@@ -254,6 +286,23 @@ private static Stream GetResourceStream(string resourceName)
                 generatedLocation: new MappingLocation(generatedLocation, contentLength));
         }
 
+        private class PathValidatingChunkInheritanceUtility : ChunkInheritanceUtility
```

Comment:
The only piece that gets the source file name where the path being relative is important is the `ChunkInheritanceUtility` so adding this class is necessary. The other alternative was adding a semi-unit semi-functional test which took the entire route of resolving a view based on a rooted path; to fully test that approach without exploding took a decent amount of work/replacement.